core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -240,8 +240,10 @@ private[spark] class TaskSchedulerImpl private[scheduler](
         // 2. The task set manager has been created but no tasks has been scheduled. In this case,
         //    simply abort the stage.
         tsm.runningTasksSet.foreach { tid =>
-          val execId = taskIdToExecutorId(tid)
-          backend.killTask(tid, execId, interruptThread, reason = "stage cancelled")
+          if (taskIdToExecutorId.contains(tid)) {
+            val execId = taskIdToExecutorId(tid)
+            backend.killTask(tid, execId, interruptThread)
```

Comment:
I think you lost the value of "reason" here.
It's not a big deal but you could also write:
```
taskIdToExecutorId.get(tid).foreach(execId => backend.killTask(tid, execId, interruptThread, reason = "stage cancelled"))